sklearn/base.py

```diff
@@ -244,7 +244,7 @@ def __repr__(self):
 
     def __getstate__(self):
         try:
-            state = super(BaseEstimator, self).__getstate__()
+            state = super().__getstate__()
```

Comment:
Yes that might be true. I just thought that would be the reason for the pickle tests failing, but apparently not. I will need to have a closer look at this.

Reply:
Actually, it does not seem to matter, another line handling masked arrays for numpy =< 11 was causing the issue. I still find it  hard to wrap my head around `super` , but you can verify that these two give the same result.

```python
class A(object):
    def print(self):
        print(super())
        print('A')
        
class B(A):
    def print(self):
        super().print()
        print('B')
        
class C(B):
    pass

class D(C):
    pass

B().print()

C().print()

D().print()
```

```python
class A(object):
    def print(self):
        print('A')
        
class B(A):
    def print(self):
        super(B, self).print()
        print('B')
        
class C(B):
    pass

class D(C):
    pass

B().print()

C().print()

D().print()
```

Reply:
ok, thanks for investigating.

Reply:
Yes,

> One big problem with 'super' is that it sounds like it will cause the superclass's copy of the method to be called. This is simply not the case, it causes the next method in the MRO to be called.

so it should be fine.